lib/rubocop/cop/style/ip_addresses.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'resolv'
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for IP addresses in strings
+      class IpAddresses < Cop
+        include StringHelp
+
+        IPV6_MAX_SIZE = 45 # IPv4-mapped IPv6 is the longest
+        MSG = 'Do not hardcode IP addresses.'.freeze
+
+        def offense?(node)
+          contents = node.source[1...-1]
+
+          return false if whitelist.include?(contents.downcase)
+
+          # To try to avoid doing two regex checks on every string,
+          # shortcut out if the string does not look like an IP address
+          return false unless could_be_ip?(contents)
```

Comment:
I am assuming that a codebase would have a lot of strings, which means that rubocop would end up doing a lot of regex matching. In order to try to prevent that, I am doing some basic checks that the string kind of looks like an IP and short-circuiting if not.

I'm not 100% sure this is necessary or the right way to go about this, so I'd love feedback here.